pkg/operation/botanist/component/resourcemanager/resource_manager.go

```diff
@@ -551,6 +574,16 @@ func (r *resourceManager) computeCommand() []string {
 	if r.values.MaxConcurrentHealthWorkers != nil {
 		cmd = append(cmd, fmt.Sprintf("--health-max-concurrent-workers=%d", *r.values.MaxConcurrentHealthWorkers))
 	}
+	if r.values.MaxConcurrentRootCAPublisherWorkers != nil {
+		cmd = append(cmd, fmt.Sprintf("--root-ca-publisher-max-concurrent-workers=%d", *r.values.MaxConcurrentRootCAPublisherWorkers))
+	}
+	if r.values.MaxConcurrentRootCAPublisherWorkers != nil {
+		if r.secrets.RootCA != nil {
+			cmd = append(cmd, fmt.Sprintf("--root-ca-file=%s/%s", volumeMountPathRootCA, secrets.DataKeyCertificateCA))
+		} else {
+			cmd = append(cmd, fmt.Sprintf("--root-ca-file=%s/ca.crt", volumeMountPathAPIServerAccess))
+		}
+	}
```

Comment:
I need help if I understood this logic:
Case A) `r.secrets.RootCA != nil`  (for example when deploying the grm into the shoot ns of the Seed targeting the [Shoot](https://github.com/gardener/gardener/blob/5367fb4d491c5b610e9c278ef368d974a7be016c/pkg/operation/botanist/resource_manager.go#L82)). The provided secret is then mounted to the container.
 
Case B) `r.secrets.RootCA == nil`(for example when deploying it into the garden ns of the Seed targeting the Seed KAPI) then we are using the `volumeMountPathAPIServerAccess` which points to the ServiceAccount. This works as for targeting the Seed we can use the projected volume token.

Is my understanding correct?
Maybe we can make that a bit more understanding by adding a comment like
```
# default to using the ca from the projected Service Account token. Relevant when NOT targeting another cluster but publishing the ca of the cluster the gardener-resource-manager runs in.
cmd = append(cmd, fmt.Sprintf("--root-ca-file=%s/ca.crt", volumeMountPathAPIServerAccess))
```
Above is a bit verbose but a comment in that direction would have helped me

Reply:
Yes, your understanding is correct. I've added a comment as suggested.